### PR TITLE
fix: implement message-port ipfs.ls

### PR DESCRIPTION
**Pull request by Gozala**
This implements `ipfs.ls` for the `ipfs-message-port-*` libraries as per https://github.com/ipfs/js-ipfs/issues/3252

### CODE REVIEW

File: packages/ipfs-message-port-server/src/ipfs.ts [hunk 1]
```diff
@@ -120,13 +127,15 @@ interface LsOptions extends AbortOptions {
...
+export type LsEntry = {
   name: string
+  path: string
   type: FileType
   size: number
+  depth: number
   cid: CID
-  mode: Mode
-  mtime: UnixFSTime
+  mode?: Mode
```
Comment (achingbrain) [inline packages/ipfs-message-port-server/src/ipfs.ts:137]: mode is not optional as it has a [default value](https://github.com/ipfs/specs/blob/master/UNIXFS.md#metadata) for folders and files

File: packages/ipfs-message-port-server/src/core.js [hunk 2]
```diff
@@ -231,6 +230,26 @@ class CoreService {
     const content = this.ipfs.cat(location, { offset, length, timeout, signal })
     return encodeCatResult(content)
   }
+
+  /**
+   * @typedef {Object} LsResult
```
Comment (achingbrain) [inline packages/ipfs-message-port-server/src/core.js:235]: We should be able to pull the definition for this in from core the same way we're doing in the http client in #3281 right?  Probably in a follow up PR

**Review by achingbrain — CHANGES_REQUESTED** [review]

**Review by achingbrain — APPROVED** [review]

File: packages/ipfs-message-port-server/src/ipfs.ts (hunk 1 above)
Comment (Gozala) [inline packages/ipfs-message-port-server/src/ipfs.ts:137]:
```suggestion
  mode: Mode
```

Comment (Gozala) [inline packages/ipfs-message-port-server/src/ipfs.ts]:
```suggestion
  mode: Mode
```

**Pull request merged**